src/MangaJet/app/src/main/java/com/mangajet/mangajet/data/Manga.kt

```diff
@@ -1,13 +1,87 @@
 package com.mangajet.mangajet.data
 
+import org.json.JSONArray
+import org.json.JSONObject
+
 // Class that represents one specific manga, stores info about it(name, author, genre...) and chapters of this manga
 class Manga {
     val id: String               // This manga unique identifier
     val library: AbstractLibrary // Library which this manga belongs
 
+    //manga info
+    var originalName: String = ""                    // Manga's original name
+    var russianName: String = ""                     // Manga's russian name
+    var author: String = ""                          // Manga's author name
+    var tags: Array<String> = emptyArray()           // Array of manga's tags
+    var description: String = ""                     // Manga's description
+    var rating: Double = 0.0                         // Manga's rating(some libraries don't support it)
+    var cover: String = ""                           // URL to manga's cover
+    var chapters: Array<MangaChapter> = emptyArray() // Array of this manga's chapters
+    var lastViewedChapter: Int = 0                   // Index of last viewed chapter
+
     // Constructor for Libraries
     constructor(library: AbstractLibrary, id: String) {
         this.library = library
         this.id = id
     }
+
+    // Utility function to get some data from JSON string by tag nam
+    private fun getDataByTag(text: String, tag: String) : String {
+        val tagToFind = "$tag\":\""
+        var start = text.indexOf(tagToFind) + tagToFind.length
+        return text.subSequence(start, text.indexOf("\"", start)).toString()
+    }
+
+    // Function to fill all manga info except chapters
+    // MAY THROW MangaJetException
+    fun updateInfo(){
+        val mangaInfo = library.getMangaInfo(id) // Exception may be thrown here
```

Comment:
use JSONObject here. It can be constructed from mangaInfo and you won't need this parser